engine/graphics/src/opengl/graphics_opengl.cpp

```diff
@@ -2369,67 +2316,207 @@ static void LogFrameBufferError(GLenum status)
+    static void CreateRenderTargetAttachment(OpenGLContext* context, OpenGLRenderTargetAttachment& attachment, AttachmentType type, const TextureParams params, const TextureCreationParams creation_params)
     {
-        OpenGLContext* context = (OpenGLContext*) _context;
+        attachment.m_Params = params;
+        attachment.m_Type   = type;
 
-        OpenGLRenderTarget* rt = new OpenGLRenderTarget();
-        rt->m_BufferTypeFlags  = buffer_type_flags;
-        rt->m_DepthBufferBits  = context->m_DepthBufferBits;
+        switch(type)
+        {
+            case ATTACHMENT_TYPE_BUFFER:
+                glGenRenderbuffers(1, &attachment.m_Buffer);
+                CHECK_GL_ERROR;
+                break;
+            case ATTACHMENT_TYPE_TEXTURE:
+                attachment.m_Texture = NewTexture(context, creation_params);
+                break; 
+            default: assert(0);
+        }
 
-        glGenFramebuffers(1, &rt->m_Id);
-        CHECK_GL_ERROR;
-        glBindFramebuffer(GL_FRAMEBUFFER, rt->m_Id);
-        CHECK_GL_ERROR;
+        ClearTextureParamsData(attachment.m_Params);
+    }
+
+    static inline void AttachRenderTargetAttachment(OpenGLContext* context, OpenGLRenderTargetAttachment& attachment, GLenum* attachment_targets, uint32_t num_attachment_targets)
+    {
+        if (attachment.m_Attached)
+        {
+            return;
+        }
 
-        memcpy(rt->m_BufferTextureParams, params, sizeof(TextureParams) * MAX_BUFFER_TYPE_COUNT);
-        // don't save the data
-        for (uint32_t i = 0; i < MAX_BUFFER_TYPE_COUNT; ++i)
+        if (attachment.m_Type == ATTACHMENT_TYPE_BUFFER)
         {
-            rt->m_BufferTextureParams[i].m_Data = 0x0;
-            rt->m_BufferTextureParams[i].m_DataSize = 0;
+            for (int i = 0; i < num_attachment_targets; ++i)
+            {
+                glFramebufferRenderbuffer(GL_FRAMEBUFFER, attachment_targets[i], GL_RENDERBUFFER, attachment.m_Buffer);
+                CHECK_GL_ERROR;
+                CHECK_GL_FRAMEBUFFER_ERROR;
+            }
+        }
+        else if (attachment.m_Type == ATTACHMENT_TYPE_TEXTURE)
+        {
+            OpenGLTexture* attachment_tex = GetAssetFromContainer<OpenGLTexture>(context->m_AssetHandleContainer, attachment.m_Texture);
+            for (int i = 0; i < num_attachment_targets; ++i)
+            {
+                glFramebufferTexture2D(GL_FRAMEBUFFER, attachment_targets[i], GL_TEXTURE_2D, attachment_tex->m_TextureIds[0], 0);
+                CHECK_GL_ERROR;
+                CHECK_GL_FRAMEBUFFER_ERROR;
+            }
+        }
+        else assert(0);
+
+        attachment.m_Attached = true;
+    }
+
+    static void ApplyRenderTargetAttachments(OpenGLContext* context, OpenGLRenderTarget* rt, bool update_current = false)
```

Comment:
We try to stay away from default values in functions if possible.
Since it's local to this file, it should be possible to be explicit here as well?

Reply:
Yes I agree, I think it was old code that was just renamed and moved. I'll fix it